lodash/lodash-tests.ts

```diff
@@ -2245,6 +2245,46 @@ namespace TestUniqBy {
     }
 }
 
+// _.uniqWith
+namespace TestUniqWith {
+    type SampleObject = {a: number; b: string; c: boolean};
+
+    let array: any[];
+    let objArray: SampleObject[];
+    let letters: string;
+    let comparator: (a: any, b : any) => boolean;
+
+    {
+        let result: any[];
+        result = _.uniqWith(array, comparator);
+    }
+
+    {
+        let result: SampleObject[];
+        result = _.uniqWith(objArray, comparator);
+    }
+
+    {
+        let result: string[];
+        result = _.uniqWith('abc', comparator);
+    }
+
+    {
+        let result: _.LoDashImplicitArrayWrapper<string[]>;
+        result = _(['abc', 'bca']).uniqWith<string[]>(comparator);
+    }
+
+    {
+        let result: _.LoDashImplicitArrayWrapper<string[]>;
+        result = _('abc').uniqWith<string[]>(comparator);
+    }
+
+    {
+        let result: _.LoDashImplicitArrayWrapper<string[]>;
+        result = _('abc').chain().uniqWith<string[]>(comparator);
```

Comment:
This is type arguments but we enforce it. 

Ideally TS bases on Typings should be able to understand true that the result of `_(<string>).chain().uniqWith(comparator)` is `_.LoDashImplicitArrayWrapper<string[]>` 

Is TS typings system capable of handling such cases? If it is then probably it is required to improve typings of `chain` or `_`.